test/compress/properties.js

```diff
@@ -1100,3 +1100,39 @@ const_prop_assign_pure: {
         x();
     }
 }
+
+join_object_assignments: {
+    options = {
+        evaluate: true,
+        join_vars: true,
+    }
+    input: {
+        console.log(function() {
+            var x = {
+                a: 1,
```

Comment:
Can you throw in a prop here with a side effect?
```
c: (console.log("c"), "C"),
```

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2763/commits/ab0fa07e1b2398fcdd1554c2893cc87a2cbe55d9